File: array_string/_go/palindrome_string.go
package main

import "fmt"

func isPalindrome(A string) int {
	var arr []int32
	for _, val := range A {
		if (val >= 48 && val <= 57) || (val >= 65 && val <= 90) || (val >= 97 && val <= 122) {
			if val >= 65 && val <= 90 {
				val += 32
				arr = append(arr, val)
			} else {
				arr = append(arr, val)
			}
		}
	}
	i, j := 0, len(arr)-1
	for i <= j {
		if arr[i] != arr[j] {
			return 0
		}
		i++
		j--
	}
	return 1
}

func main() {
	A := "A man, a plan, a canal: Panama"
	//A := "1a2"
	fmt.Println(isPalindrome(A))
}

File: array_string/Palindrome_String.py
# https://www.interviewbit.com/problems/palindrome-string/?study_plan=study-plan-3-months&/

def isPalindrome(A: str) -> int:
    arr = [x.lower() for x in A if x.isalnum()]
    i, j = 0, len(arr) - 1
    while i <= j:
        if arr[i] != arr[j]:
            return False
        i += 1
        j -= 1
    return True


A = "A man, a plan, a canal: Panama"
# A = "Nas"
print(isPalindrome(A))
